### DISCUSSION TITLE
Wildcard in validation $rules (Livewire Array)

### DISCUSSION THREAD

**Discussion by nam-co** [livewire/livewire#5542] [1. Help] [answered]
Hi, hope somebody can help me, Im getting a error: Cannot bind property [page.names.en] without a validation rule present in the [$rules] array on Livewire component:

View
```
          @foreach ($account->languages as $lang)
          <figure>
              <label>{{$lang}}</label>
              <input type="text" wire:model.lazy="page.names.{{$lang}}">
          </figure>
          @endforeach
//THIS SHOWS
              <input type="text" wire:model.lazy="page.names">
```

Component rules
```
    protected $rules = [
       "page.names.*" => "sometimes|string|max:100",
    ];
```

What Im doing wrong?  I tried adding to rules 'page.names.en' on mount, but it dint work

**Comment by joshhanley** [comment]
@nam-co can you confirm what exactly `page.names` is? Is `page` an Eloquent model with an attribute called `names` that is an array of data? If so, I think you may need to whitelist `page.names`, but if that doesn't work, you may need two sets of rules `page.names` and `page.names.*`.

The binding should be your first example `wure:model.lazy="page.names.{{ $lang }}"`

Hope this helps!

**Reply by nam-co** [reply]
Hi @joshhanley, thanks for answering, page.names is a json column, so it will hold something like {"en": "Hello","es": "Hola"} 

I tried putting both page.names and page.names.*, but I still get:  "Cannot bind property [page.names.en] without a validation rule present in the [$rules] array on Livewire component"

**Reply by joshhanley** [reply]
@nam-co ah yeah, JSON columns can be problematic. Yeah, sometimes it's easier to copy the data to a public property and use it from there instead of binding directly to the model attribute, especially if the attribute is json/array.

**Comment by nam-co** [comment, marked as answer]
For anyone looking for this answer, In V2 the problem only shows when objects are involved, with arrays it works

**Comment by marcdegagne** [comment]
I know this is marked as answered but this may help.
Not sure if this would answer exactly the issue.
I was able to get a single instance validated and get the error message but only as a general message only. Not for the individual instances.

**The view:**
```blade
@foreach( $habitats as $key => $habitat )
      <div class="">
          <div wire:key="habitats.{{ $key }}" class="">
              <input wire:model.lazy="habitats.{{ $key }}.type" type="text" class="" >
              <button wire:click="deleteHabitat( {{ $habitat->id }} )" class=""></button>
          </div>
      </div>
@endforeach
```

**The properties:**
```php
public string $type;

public Collection $habitats;
```


**The rules:**
```php
 protected $rules = [
       // The first one is required otherwise the values wont show in the view at mount time (not sure why).
       "habitats.*.type"    => "required|regex:/^[\pL\s\-']+$/u|unique:habitats,type|max:50",
       // This is the one required for the single instance validation
       "type"    => "required|regex:/^[\pL\s\-']+$/u|unique:habitats,type|max:50",
 ];
```


**The updated function:** 
```php
 public function updatedHabitats( $type_value, $key )
    {
        # dd( $type_value, $key );  // the value, 0.type

        $index = explode( ".", $key, -1 )[0];
        $this->type = $this->habitats[$index]->type;
        $this->validateOnly( "type" );

        // The rest of code
}
```
You can try the `public function updating( $fullkey, $value )` or the `public function updated( $fullkey, $value )` also.
These inject two parameters also but you get the full key used in the view as for me this translate to `habitats.0.type`.
The `updatedYourAttribute()`method uses first the $value and after the partial $Key (*.type).
